src/js/components/Button/Button.js

```diff
@@ -258,6 +331,70 @@ const Button = forwardRef(
       contents = first || second || children;
     }
 
+    const defaultBadgeDimension =
+      typeof badgeProp === 'boolean' ||
+      (badgeProp && badgeProp.value && typeof badgeProp.value === 'boolean')
+        ? // empty badge should be smaller. this value was chosen as a default
+          // after experimenting with various values
+          `${parseInt(theme.button.badge.size.medium.replace('px', ''), 10) /
+            2}px`
+        : theme.button.badge.size.medium;
+
+    const [badgeWidth, setBadgeWidth] = useState(defaultBadgeDimension);
```

Comment:
Would it work to manage this as `badgeDimension`, which would be `[width, height]`? Having them separate results in a bit more duplication. Just a thought.

Reply:
Good idea. let me clean that up.